File: week6/mario/more/mario.py
# the following program prints half of a pyramid of a given height h
# definition of functions:
def get_height():  # ensures that the user introudces an integer
    height = ""        # between 0 and 9
    while (True):
        try:
            height = int(input("height: "))
            if height <= 0 or height >= 9:
                continue
        except ValueError:  # if ValueError is raised
            continue       # the user gets asked for input again
        else:
            break
    return height


def print_row(height, row):  # print the ith row for a pyramid of height h
    for column in range(height):  # if we consider the whole structure as a square
        if column < row:        # for the ith row there will be i spaces and h-i
            print(" ", end="")  # hashes. Columns are numerated from left to right
        else:
            print("#", end="")
    print("  ", end="")         # separation between pyramids
    for column in range(height - row):  # for the second pyramid,
        print("#", end="")      # a row will have (height - row) hashes
    print()                     # given the numeration used for columns/ rows


if __name__ == "__main__":  # main section of the code

    height = get_height()  # ask for the height

    for row in range(height-1, -1, -1):  # print each row
        print_row(height, row)         # rows are numbered top to bottom
    # range(height-1, -1, -1) gives me a list from height-1 to 0

File: week1/mario/less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int height, int row);
int main(void)
{
    int height;
    do
    {
        // prompt user for the pyramid height between 1 and 8 (inclusive)
        height = get_int("height: ");
    }
    while (height < 1 || height > 8);
    // loop through each row from top (row h) to bottom (row 1)
    for (int row = height; row > 0; row--)
    {
        print_row(height, row); // prints the current row of the pyramid
    }
}
void print_row(int height, int row)
{
    // prints a specific row of the double pyramid based on height and row number
    for (int j = 1; j <= height; j++) // prints spaces and hashed to make the left pyramid
    {
        if (j < row)
        {
            printf(" ");
        }
        else
        {
            printf("#");
        }
    }
    printf("\n"); // moves to the next line for the following row
}
